frontend/src/components/LabelledNumberInput.vue

```diff
@@ -123,12 +132,60 @@ export default defineComponent({
 	computed: {
 		increasable(): boolean {
 			if (this.value == null) return true;
-			return isNaN(this.value) || this.value < this.max;
+			return isNaN(this.value) || this.value < this.maximum;
 		},
-
 		decreasable(): boolean {
 			if (this.value == null) return true;
-			return isNaN(this.value) || this.value > this.min;
+			return isNaN(this.value) || this.value > this.minimum;
+		},
+		isSteppable(): boolean{
+			return this.steps != null && this.steps.length > 0
+		},
+		minimum(): number{
+			if (this.isSteppable){
+				return this.steps![0]
+			} else {
+				return this.min
+			}
+		},
+		maximum(): number{
+			if (this.isSteppable){
+				return this.steps![this.steps!.length-1]
+			} else {
+				return this.max
+			}
+		},
+		nearestStepDown(): number{
+			if (!this.isSteppable) return this.value
```

Comment:
I know it doesn't technically matter since nearestStepDown would not be used without isSteppable, but it makes more sense for the default to be this.step